### PR TITLE
config.get -> config.get_value / extra config declarations

### PR DESCRIPTION
**Pull request by smotornyuk**
❗  contains a breaking change

* Change a couple of `config.get` to `config.get_value` 
* Add a few missing config declarations

### Breaking change

Before, `toolkit.aslist` wrapped any value other than str/list/tuple into a list. I.e `aslist({1,2}) -> [{1,2}]`. Now iterables will be converted to the list, i.e `aslist({1,2}) -> list({1,2}) -> [1,2]`. Non-iterables behave in the same way: `aslist(1) -> [1]` 

Not sure if such a change is acceptable. But documentation of this method has no mention of types other than `str`, so there were no guarantees, right? :)
In this particular case, we need this change because of the `WTF_CSRF_METHODS` config options. It's declared as a list, but internally flask-wtf converts it into the set. *If* validators are applied to this converted value again(it generally happens only in tests, but it's a possible case for the real world as well and I even mentioned in in [the docs](https://docs.ckan.org/en/latest/maintaining/configuration.html#declaring-config-options) - check first note about idempotence of validators), we'll get `aslist({"POST", "PUT"}) -> [{"POST", "PUT"}]`. And then we'll get a type error when `flask-wtf` try to convert this value into a set once again.

*We can* create a separate validator or even declaration type(i.e, allow using something like `type: set` in the declaration file), but:
* It's really a rare case. Even though I would choose it, there is a second reason:
* We are releasing a new CKAN version. It's the best time to introduce breaking changes. And this particular change makes code more intuitive, isn't it?

ℹ️ Tests are fixed in #7254

### CODE REVIEW

File: changes/7257.feature [hunk 1]
```diff
@@ -0,0 +1,2 @@
+`toolkit.asbool` now converts any iterable other than `list` and `tuple` into a `list`: `list(value)`.
```
Comment (amercader) [inline changes/7257.feature:1]:
```suggestion
`toolkit.aslist` now converts any iterable other than `list` and `tuple` into a `list`: `list(value)`.
```

Also maybe it's easier to understand the change with an example

File: ckan/config/config_declaration.yaml [hunk 2]
```diff
@@ -1241,16 +1254,29 @@ groups:
...
-        example: templates
+        example: templates-bs3
         description: |
           This config option is used to configure the base folder for templates used
           by CKAN core. It is currently unused and it only accepts one value: ``templates``
           (Bootstrap 3, the default value from CKAN 2.8 onwards).
 
+      - key: ckan.default.package_type
+        default: dataset
+        description: Default type of dataset that used in UI links(eg. **New Dataset**)
```
Comment (amercader) [inline ckan/config/config_declaration.yaml:1265]:
```suggestion
        description: Default type of dataset that will be used in the UI links (eg. "New Dataset")
```
Maybe an example would make this clear. Does it mean that if `ckan.default.package_type=record`, the UI will show "New Record"? And the same for groups and orgs

**Review by amercader — COMMENTED** [review]
@smotornyuk I think it's a reasonable breaking change to introduce if we document it properly in the changelog. I just added some comments to make clearer the changelog entry and the config docstrings

File: ckan/config/config_declaration.yaml (hunk 2 above)
Comment (smotornyuk) [reply ckan/config/config_declaration.yaml:1265]: Yes, exactly. If you have a `record` package type defined, of course. I and Ian added this functionality 2 years ago. #5337 contains details for the group/organization. The package entity behaves in the same manner. 
This feature potentially provides an easy way to solve issues like #7200 or #5331, where people insist, that the "group" term does not reflect the idea of a group entity. Now they can just create a custom group type and set it as a default one. For non-standard types, that cannot be easily transformed into a human-readable label, chaining [`humanize_entity_type`](https://github.com/ckan/ckan/blob/master/ckan/lib/helpers.py#L1002) helper can solve a problem.

**Comment by smotornyuk** [comment]
I accepted your suggestions and added another couple of lines about the breaking change and `default.package_type` option

**Pull request merged**